lib/XMLHttpRequest.js

```diff
@@ -374,7 +374,8 @@ function XMLHttpRequest(opts) {
     } else if (data) {
       headers["Content-Length"] = Buffer.isBuffer(data) ? data.length : Buffer.byteLength(data);
 
-      if (!headers["Content-Type"]) {
+      var headersKeys = Object.keys(headers);
+      if (!headersKeys.some(h => /content-type/i.test(h))) {
```

Comment:
This module should remain compatible with some extremely archaic Node.js versions, we can therefore not rely on ES2015 features (such as fat-arrow functions) being supported.

Also, I wouldn't use a `RegExp` here for the simple reason that it's more difficult to ensure it's limited to only the header you're looking for. Your current `RegExp`, for instance, also matches `"X-Content-Type"`. `RegExp` functionality also isn't really required here.

All in all I'd change it to:

```suggestion
      if (!headersKeys.some(function (h) { return h.toLowerCase() === 'content-type' })) {
```